torch/_lobpcg.py

```diff
@@ -563,10 +563,19 @@ def _get_rayleigh_ritz_transform(self, S):
         SBS = _utils.qform(B, S)
         d_row = SBS.diagonal(0, -2, -1) ** -0.5
         d_col = d_row.reshape(d_row.shape[0], 1)
-        R = torch.cholesky((SBS * d_row) * d_col, upper=True)
-        # TODO: could use LAPACK ?trtri as R is upper-triangular
-        Rinv = torch.inverse(R)
-        return Rinv * d_col
+        # Rt := R^T
+        Rt = torch.cholesky((SBS * d_row) * d_col, upper=False)
```

Comment:
What is the reason for the `upper=False`? It is more efficient for the `triangular_solve` below?

Reply:
Exactly, the we need inverse to be to the left from `d`, not to the right.